### PR TITLE
ROX-23132: use make tag result in image names on Konflux

### PR DESCRIPTION
**Pull request by tommartensen**
## Description

[Build pipeline](https://console.redhat.com/preview/application-pipeline/workspaces/rh-acs/applications/acs/pipelineruns/central-db-on-pull-request-z7qjg) produced `quay.io/redhat-user-workloads/rh-acs-tenant/acs/central-db:4.4.x-436-gd6f46a5cdd`. 
 
I have not updated other pipelines yet, I want to have a discussion first if we should commit to this new image tagging scheme & what happens with image flavors & `-fast` suffix.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Konflux PR pipeline for `central-db` passes.

### CODE REVIEW

File: .tekton/central-db-pull-request.yaml [hunk 1]
```diff
@@ -240,10 +239,39 @@ spec:
...
+          - name: determine-image-tag
+            image: quay.io/stackrox-io/apollo-ci:stackrox-build-0.3.66
+            env:
+              - name: IMAGE
+                value: $(params.image)
+            script: |
+              #!/usr/bin/env bash
+              set -euo pipefail
+              cd "$(workspaces.source.path)/source"
+              echo -n "${IMAGE}:$(make tag)" > "$(results.image-with-tag.path)"
```
Comment (msugakov) [inline .tekton/central-db-pull-request.yaml:265]: There's a special mantra to prevent `make` from printing extra staff when just the `tag` is required. It usually isn't needed but there are weird edge cases when it helps.

```suggestion
              echo -n "${IMAGE}:$(make --quiet --no-print-directory tag)" > "$(results.image-with-tag.path)"
```

File: .tekton/central-db-pull-request.yaml [hunk 2]
```diff
@@ -240,10 +239,39 @@ spec:
...
+              cat "$(results.image-with-tag.path)"
```
Comment (msugakov) [inline .tekton/central-db-pull-request.yaml:266]: Suggesting `echo ... | tee blah` instead of `echo ... > blah; cat blah` if the idea is to show the contents.

**Review by msugakov — COMMENTED** [review]
Made a quick shallow look. Please expect more comments in further iterations.

File: .tekton/central-db-pull-request.yaml [hunk 3]
```diff
@@ -240,10 +243,39 @@ spec:
       - name: source
         workspace: workspace
 
+    - name: determine-image-tag
+      params:
+        - name: image
+          value: $(params.output-image)
+      runAfter:
+      - clone-repository
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml:228]: Reordered.

[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml:228]: This didn't occur in `central-db`, because there are no prefetches, but I see the point.

File: .tekton/central-db-pull-request.yaml [hunk 4]
```diff
@@ -240,10 +243,39 @@ spec:
...
+      taskSpec:
+        params:
+          - name: image
+            description: Image without tag
+        results:
+          - name: image-with-tag
+            description: Fully Qualified Image Name with identifying tag
+        steps:
+          - name: determine-image-tag
+            image: quay.io/stackrox-io/apollo-ci:stackrox-build-0.3.66
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml:238]: The things I tried here:
- ❌ use a plain ubi image and install `make` & `git` - after sifting through installation errors, `go` was still missing.
- ❌ used `openshift-golang-builder` (has `make`, `git`, `go` installed): Because it is Brew based, it could not be pulled: https://github.com/stackrox/stackrox/runs/23823234740. -> might be possible to do, but the Task would need pull secrets.
- ✅ `stackrox-build` image is public, contains the necessary binaries.

[the remark this answers is not in the document]
Comment (msugakov) [reply .tekton/central-db-pull-request.yaml:238]: `quay.io/stackrox-io/apollo-ci` is not good because it was not produced by Konflux or downstream. Meaning that it creates (higher) supply chain security risk since it's built outside of systems that are considered safe.

In my opinion, the fact that `build-container` task "automagically" uses pull secrets but those aren't "automagically" used at the pipeline level is worth reporting as an improvement request to the Konflux team. If these are pull secrets, they should ideally work consistently in different places of the pipeline.

I see two ways to approach the supply chain security:
1. There's nothing in the `Makefile` that _requires_ `go` command for `tag` target, I believe. It's just a coincidence that some `go` commands run when the `Makefile` gets interpreted. There might be a way to refactor it so that `go` is not a requirement for `make tag`.
2. Since we have a pull secret for `brew.registry.redhat.io`, you can see a way to use it in the task.

[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml]: I went with option 1 and moved the `tag` target to a separate Makefile, which we call directly in Konflux and include in the main Makefile. I am relying on successful GHA builds & e2e tests to verify the absence of regressions. 

Using a pull secret was not possible, because:
- TaskRuns (in Tekton PipelineRuns or standalone) use a service account that has access to pull secrets
  -  there are two interesting ones in our workspace, both of which have pull and mountable secrets for each component in Konflux:
      1. `default` 
      1. `appstudio-pipeline` - in addition to above, can _mount_ our custom secrets for brew, registry.redhat.io, quay.io/rhacs-eng -> these are used in buildah task.
    - AFAICT there was no option to add the brew secret as an image pull secret to the service account (without asking Konflux team, who may or may not chose to do this). Also, we're not allowed to create new SAs in the namespace, so cannot craft one that has the required imagepullsecrets as we like.

[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml]: One follow-up (also to satisfy the EC requirements for task bundles) is to turn the inline task definition into a proper task, which is bundled and can be pulled from a trusted source. This is already captured in https://issues.redhat.com/browse/ROX-22196 .

[the remark this answers is not in the document]
Comment (msugakov) [reply .tekton/central-db-pull-request.yaml]: Regarding secrets.

Disclaimer: I prefer the approach with the `Makefile` and so the question about secrets is for my education only.

Can you please explain or point me to docs that tell what are _mountable_ secrets? I wonder how these work and under which conditions they get mounted. Asking because I could not (obviously) spot `buildah login` commands when scrolling through buildah task and for me it's total mystery how image pull secrets become available for builds. Perhaps this information will help me realize why this does not happen for pulling containers in a custom task.

----

Regarding ROX-22196.

ROX-22196 is currently an umbrella ticket for any custom task. This is fine but Ralph Bean [expressed an interest](https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1713188634789549?thread_ts=1713181360.407219&cid=C04PZ7H0VA8) to have image tag generation task contributed to <https://github.com/redhat-appstudio/build-definitions>. While we could create a task for ourselves, contributing it to community sounds really cool (although, again, not mandatory).

For this idea to not be lost, it should be captured somewhere. Right now, neither code nor ROX-22196 tells us about it. It would be nice if you create a dedicated ticket for factoring out image tag determination task and mention redhat-appstudio/build-definitions opportunity there.

[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/central-db-pull-request.yaml]: > Can you please explain or point me to docs that tell what are mountable secrets?

I think these are good explanations:
* https://livebook.manning.com/concept/kubernetes/mountable-secret#:~:text=Understanding%20a%20ServiceAccount's%20mountable%20secrets&text=By%20default%2C%20a%20pod%20can,Secrets%20on%20the%20Service%2DAccount.
* https://stackoverflow.com/a/72139678
* https://tekton.dev/docs/pipelines/auth/#basic-auth-for-docker

TLDR: mountableSecrets: An allowlist for which secrets a service account can mount and use to talk to K8s API. 

`buildah pull` (https://github.com/redhat-appstudio/build-definitions/blob/main/task/buildah/0.1/buildah.yaml#L182-L186) looks for authentication files: 

> Path of the authentication file. Default is ${XDG_RUNTIME_DIR}/containers/auth.json. See containers-auth.json(5) for more information. This file is created using buildah login.
> If the authorization state is not found there, $HOME/.docker/config.json is checked, which is set using docker login.
> From: https://github.com/containers/buildah/blob/main/docs/buildah-pull.1.md
 
I don't know where it is done, but the taskRuns of `build-container` task contain the `serviceAccountName: appstudio-pipeline` annotation. The resulting pods mount all the secrets of the service account (ie including the custom pull secrets) to a place that buildah seems to understand. They're not K8s imagePullSecrets, because the pulling of the base images takes place inside the task/pod, not on K8s level. 

---
> Regarding [ROX-22196](https://issues.redhat.com/browse/ROX-22196).

Added https://issues.redhat.com/browse/ROX-23812 as split from ROX-22196 and a comment in the code.

File: .tekton/central-db-pull-request.yaml [hunk 5]
```diff
@@ -221,12 +220,43 @@ spec:
       - name: basic-auth
         workspace: git-auth
 
+    - name: determine-image-tag
```
Comment (msugakov) [inline .tekton/central-db-pull-request.yaml:223]: Sorry, once again regarding the placement of this task in the pipeline. Yann's PR
https://github.com/stackrox/stackrox/pull/10851 made me look again at the [init task](https://github.com/redhat-appstudio/build-definitions/blob/main/task/init/0.2/init.yaml) and I realized that `init` should go _after_ we determine the tag.
For us right now it does not matter since we want to always rebuild and the wrong tag that `init` checks does not play the role, but it may matter in some future when colleagues add more responsibilities to `init` and/or when we contribute `determine-image-tag`.

Again, no ready solution for that on my side. Do you have any thoughts how to best approach it?

File: Makefile [hunk 6]
```diff
@@ -26,9 +26,7 @@ SILENT ?= @
 # TODO: [ROX-19070] Update postgres store test generation to work for foreign keys
 UNIT_TEST_IGNORE := "stackrox/rox/sensor/tests|stackrox/rox/operator/tests|stackrox/rox/central/reports/config/store/postgres|stackrox/rox/central/auth/store/postgres|stackrox/rox/scanner/e2etests"
 
-ifeq ($(TAG),)
-TAG=$(shell git describe --tags --abbrev=10 --dirty --long --exclude '*-nightly-*')
-endif
+include make/tag.mk
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply Makefile]: Not sure we are on the same page about the importance of this issue. 

I reverted the Makefile changes, as the image tag is produced correctly and the `go: command not found` stderr can only be seen in the log. 

So how do you suggest to address `GOGO_DIR` and `GRPC_GATEWAY_DIR`?
Sure we could add a check to see if `go` is installed, but won't this break other workflows?

File: .tekton/central-db-pull-request.yaml (hunk 5 above)
Comment (msugakov) [reply .tekton/central-db-pull-request.yaml]: > in some future when colleagues add something to `clone-repository` that requires `init` to have been called (eg. placing some contents in the workspace source before the checkout)

To me this feels made up because `init` does not attach any workspace, but nevertheless I share the uncertainty about `init` because it is ambiguously described and to me it isn't clear _what `init` will **not** be_ responsible for.

It's own description
> Initialize Pipeline Task, include flags for rebuild and auth. Generates image repository secret used by the PipelineRun.

does not help, and besides the image repository secret part is a lie according to [its own source](https://github.com/redhat-appstudio/build-definitions/blob/82f7008ec8ecbfd8f4c1218fff3924d97730eb71/task/init/0.2/init.yaml#L37-L48).

I don't feel we're standing on a solid ground with the Konflux team sharing our vision. Therefore, I think the safest is to align on our use with the Konflux team, as we discussed in [today's sync](https://docs.google.com/document/d/1bCKSof0166b5M0e9hvgMBgAPzU9-jH-dEHDTjqrHd54/edit#heading=h.m9hruc80pomn).

File: .tekton/basic-component-pipeline.yaml [hunk 7]
```diff
@@ -172,12 +172,45 @@ spec:
         - name: basic-auth
           workspace: git-auth
 
+    - name: determine-image-tag
+      params:
+        - name: image
+          value: $(params.output-image)
+      runAfter:
+      - clone-repository
```
Comment (msugakov) [inline .tekton/basic-component-pipeline.yaml:180]: [nit] A small comment here that the task must run on a freshly cloned repository to prevent seeing any changes from other tasks would be helpful.

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (msugakov) [reply Makefile]: > Not sure we are on the same page about the importance of this issue.
I'm not sure our understanding of _what is_ the issue was the same and changed the same way throughout the life of this PR :wink:

I think, taking the debt of `go: command not found` lines in `stderr` is perhaps the least evil. Fixing GOGO_DIR and GRPC_GATEWAY_DIR alone won't be enough because SCANNER_DIR would still produce these errors.

File: .tekton/basic-component-pipeline.yaml [hunk 8]
```diff
@@ -172,12 +172,45 @@ spec:
...
+          - name: image-with-tag
+            description: Fully Qualified Image Name with identifying tag
+        steps:
+          - name: determine-image-tag
+            image: registry.access.redhat.com/ubi8:latest
+            env:
+              - name: IMAGE
+                value: $(params.image)
+              - name: KONFLUX_CI
+                value: 'true'
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml:196]: The reason I included the change to the `BUILD_TAG` in this PR is that it sets the version for which the bundle is generated, ie one step closer to deployments without overrides. I see your point that this needs more discussion and will revert.

File: .tekton/basic-component-pipeline.yaml [hunk 9]
```diff
@@ -25,7 +25,7 @@ spec:
         - name: git-url
           value: $(tasks.clone-repository.results.url)?rev=$(tasks.clone-repository.results.commit)
         - name: image-url
-          value: $(params.output-image)
+          value: $(tasks.build-container.results.IMAGE_URL)
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml:28]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/e2a34d52edee3087903c2817ab948c07eb19623f

File: .tekton/basic-component-pipeline.yaml [hunk 10]
```diff
@@ -229,7 +262,7 @@ spec:
     - name: build-source-image
       params:
         - name: BINARY_IMAGE
-          value: $(params.output-image)
+          value: $(tasks.determine-image-tag.results.image-with-tag)
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/e2a34d52edee3087903c2817ab948c07eb19623f

File: .tekton/basic-component-pipeline.yaml (hunk 7 above)
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/e2a34d52edee3087903c2817ab948c07eb19623f

File: .tekton/basic-component-pipeline.yaml [hunk 11]
```diff
@@ -172,12 +172,45 @@ spec:
...
+            env:
+              - name: IMAGE
+                value: $(params.image)
+              - name: KONFLUX_CI
+                value: 'true'
+            script: |
+              #!/usr/bin/env bash
+              set -euo pipefail
+              dnf upgrade -y --nobest
+              dnf install -y git make
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/e2a34d52edee3087903c2817ab948c07eb19623f

File: .tekton/basic-component-pipeline.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/e2a34d52edee3087903c2817ab948c07eb19623f

File: .tekton/basic-component-pipeline.yaml [hunk 12]
```diff
@@ -172,12 +172,45 @@ spec:
...
           workspace: git-auth
 
+    - name: determine-image-tag
+      params:
+        - name: image
+          value: $(params.output-image)
+      runAfter:
+      - clone-repository
+      # TODO(ROX-23812): Refactor to a task bundle
+      taskSpec:
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/983d4ec69fd0ce314e14762189c4835c7d5fac6c

File: .tekton/basic-component-pipeline.yaml [hunk 13]
```diff
@@ -172,12 +172,45 @@ spec:
         - name: basic-auth
           workspace: git-auth
 
+    - name: determine-image-tag
+      params:
+        - name: image
+          value: $(params.output-image)
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/869c2ad56c5cc862b8757e5d164607a12f90a7c0

File: .tekton/basic-component-pipeline.yaml [hunk 14]
```diff
@@ -172,12 +172,45 @@ spec:
...
+      runAfter:
+      - clone-repository
+      # TODO(ROX-23812): Refactor to a task bundle
+      taskSpec:
+        params:
+          - name: image
+            description: Image without tag
+        results:
+          - name: image-with-tag
+            description: Fully Qualified Image Name with identifying tag
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: Done in https://github.com/stackrox/stackrox/pull/10749/commits/869c2ad56c5cc862b8757e5d164607a12f90a7c0 (fixed in following commits).

File: make/tag.mk [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+TAG=$(shell git describe --tags --abbrev=10 --dirty --long --exclude '*-nightly-*')
+endif
+
+define tag-impl
+echo $(TAG)
+endef
+
+.PHONY: tag
+tag:
+	$(call tag-impl)
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply make/tag.mk]: Obsolete, custom `tag.mk` file is removed.

File: .tekton/central-db-pull-request.yaml (hunk 5 above)
Comment (msugakov) [reply .tekton/central-db-pull-request.yaml]: FTR https://redhat-internal.slack.com/archives/C05TS9N0S7L/p1715096492694009?thread_ts=1714998487.394219&cid=C05TS9N0S7L

File: .tekton/determine-image-tag-task.yaml [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+kind: Task
+metadata:
+  name: determine-image-tag
+  namespace: rh-acs-tenant
+spec:
+  description: Determining image tag with custom logic.
+  params:
+  results:
+  - name: image-tag
+    description: Image Tag determine by custom logic.
```
Comment (msugakov) [inline .tekton/determine-image-tag-task.yaml:11]:
```suggestion
    description: Image Tag determined by custom logic.
```

File: .tekton/determine-image-tag-task.yaml [hunk 17]
```diff
@@ -0,0 +1,24 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: determine-image-tag
+  namespace: rh-acs-tenant
+spec:
+  description: Determining image tag with custom logic.
```
Comment (msugakov) [inline .tekton/determine-image-tag-task.yaml:7]:
```suggestion
  description: Determines image tag with custom logic.
```

File: .tekton/main-pipeline.yaml [hunk 18]
```diff
@@ -173,6 +176,17 @@ spec:
     - name: basic-auth
       workspace: git-auth
 
+  - name: determine-image-tag
+    runAfter:
+    # This task must run on a freshly cloned repository to prevent seeing any changes from other tasks.
+    - clone-repository
+    # TODO(ROX-23812): Refactor to a task bundle
```
Comment (msugakov) [inline .tekton/main-pipeline.yaml:183]: The comment is now misplaced.
```suggestion
```

File: .tekton/determine-image-tag-task.yaml [hunk 19]
```diff
@@ -0,0 +1,24 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: determine-image-tag
+  namespace: rh-acs-tenant
+spec:
```
Comment (msugakov) [inline .tekton/determine-image-tag-task.yaml:6]: Let's move the comment here.
```suggestion
# TODO(ROX-23812): Refactor this task to a task bundle
spec:
```

File: .tekton/basic-component-pipeline.yaml [hunk 20]
```diff
@@ -126,7 +126,10 @@ spec:
   - name: init
     params:
     - name: image-url
-      value: $(params.output-image)
+      # We can't provide a real tag because of it is not known at this time.
```
Comment (msugakov) [inline .tekton/basic-component-pipeline.yaml:129]:
```suggestion
      # We can't provide a real tag because it is not known at this time.
```

File: .tekton/determine-image-tag-task.yaml [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+  steps:
+  - name: determine-image-tag
+    image: registry.access.redhat.com/ubi8:latest
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      dnf -y upgrade --nobest
+      dnf -y install git make
+      cd "$(workspaces.source.path)/source"
+      echo -n "$(make --quiet --no-print-directory tag)-fast" | tee "$(results.image-tag.path)"
```
Comment (msugakov) [inline .tekton/determine-image-tag-task.yaml:21]: I just realized that we can also run `scripts/konflux/fail-build-if-git-is-dirty.sh` to actually prevent `-dirty` suffixes on the output.

File: .tekton/scanner-v4-pipeline.yaml [hunk 22]
```diff
@@ -172,9 +175,20 @@ spec:
     - name: basic-auth
       workspace: git-auth
 
-  - name: fetch-vuln-mappings
+  - name: determine-image-tag
     runAfter:
+    # This task must run on a freshly cloned repository to prevent seeing any changes from other tasks.
     - clone-repository
+    # TODO(ROX-23812): Refactor to a task bundle
```
Comment (msugakov) [inline .tekton/scanner-v4-pipeline.yaml:182]: The comment isn't needed here.
```suggestion
```

File: .tekton/basic-component-pipeline.yaml [hunk 23]
```diff
@@ -172,12 +175,23 @@ spec:
     - name: basic-auth
       workspace: git-auth
 
+  - name: determine-image-tag
+    runAfter:
+    # This task must run on a freshly cloned repository to prevent seeing any changes from other tasks.
+    - clone-repository
+    # TODO(ROX-23812): Refactor to a task bundle
```
Comment (msugakov) [inline .tekton/basic-component-pipeline.yaml:182]: This comment would be better placed in `determine-image-tag-task.yaml`.
```suggestion
```

**Review by msugakov — COMMENTED** [review]
Cosmetics

File: .tekton/main-pipeline.yaml [hunk 24]
```diff
@@ -127,7 +127,10 @@ spec:
   - name: init
     params:
     - name: image-url
-      value: $(params.output-image)
+      # We can't provide a real tag because of it is not known at this time.
```
Comment (msugakov) [inline .tekton/main-pipeline.yaml:130]:
```suggestion
      # We can't provide a real tag because it is not known at this time.
```

File: .tekton/scanner-v4-pipeline.yaml [hunk 25]
```diff
@@ -126,7 +126,10 @@ spec:
   - name: init
     params:
     - name: image-url
-      value: $(params.output-image)
+      # We can't provide a real tag because of it is not known at this time.
```
Comment (msugakov) [inline .tekton/scanner-v4-pipeline.yaml:129]:
```suggestion
      # We can't provide a real tag because it is not known at this time.
```

File: .tekton/determine-image-tag-task.yaml (hunk 17 above)
Comment (msugakov) [reply .tekton/determine-image-tag-task.yaml:7]: After seeing the description show up in the pipeline, I have a desire for better
image

How about this?
> Determines the tag for the output image using the StackRox convention (from `make tag` output).

File: .tekton/scanner-v4-pipeline.yaml (hunk 22 above)
Comment (tommartensen) [reply .tekton/scanner-v4-pipeline.yaml]: moved to task spec.

File: .tekton/basic-component-pipeline.yaml (hunk 20 above)
Comment (tommartensen) [reply .tekton/basic-component-pipeline.yaml]: fixed

File: .tekton/determine-image-tag-task.yaml (hunk 19 above)
Comment (tommartensen) [reply .tekton/determine-image-tag-task.yaml:7]: Moved here.

File: .tekton/main-pipeline.yaml (hunk 24 above)
Comment (tommartensen) [reply .tekton/main-pipeline.yaml]: fixed

File: .tekton/determine-image-tag-task.yaml (hunk 17 above)
Comment (tommartensen) [reply .tekton/determine-image-tag-task.yaml]: fixed - https://github.com/stackrox/stackrox/pull/10749/commits/d124162924490686bc76f7a3d5fef97f860af434#diff-4fb9379bf71bb227cba8211a3ebc5a3914856b3dc08b02fc311f274ba9bec32eR8.

File: .tekton/determine-image-tag-task.yaml (hunk 21 above)
Comment (tommartensen) [reply .tekton/determine-image-tag-task.yaml:23]: Added in https://github.com/stackrox/stackrox/pull/10749/commits/fe9f69ba3b36e372b39bd5d853a02889644502b3 before the `make tag` call.

File: .tekton/determine-image-tag-task.yaml (hunk 16 above)
Comment (tommartensen) [reply .tekton/determine-image-tag-task.yaml]: fixed

**Comment by tommartensen** [comment]
- [roxctl-on-pull-request-lpd7n](https://console.redhat.com/preview/application-pipeline/workspaces/rh-acs/applications/acs/pipelineruns/roxctl-on-pull-request-lpd7n) failed with a `502 Bad Gateway` while pulling `registry.access.redhat.com/ubi8/ubi-minimal:latest`. There have been successful pipeline runs for this component before, so we can ignore this as a flake.

**Review by msugakov — APPROVED** [review]
Awesome!
